ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.component.js

```diff
@@ -197,4 +54,119 @@ export default class ConfirmAddSuggestedToken extends Component {
     });
     return duplicates.length > 0;
   }
+
+  function checkTokenDuplicates() {
+    const pending = suggestedAssets.map(({ asset }) =>
+      asset.address.toUpperCase(),
+    );
+    const existing = tokens.map((token) => token.address.toUpperCase());
+    const dupes = pending.filter((proposed) => {
+      return existing.includes(proposed);
+    });
+
+    return dupes.length > 0;
+  }
+
+  function getTokenName(name, symbol) {
```

Comment:
[req] this one could remain using `function` but be elevated outside of the component as it has no dependencies on props or state.

Reply:
Yes to moving this function with no dependencies out to avoid creating a new function for every component instance! Great catch!